scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
```

Comment:
I mean that URLs extracted by LinkExtractor are then passed to Request, and it calls `safe_url_string` again. #833 is about not calling `safe_url_string` there, but if link extractor calls it anyways then it won't work for urls extracted by link extractors.